client/my-sites/patterns/components/page-view-tracker.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import getUserSetting from 'calypso/state/selectors/get-user-setting';
+
+type PatternsPageViewTrackerProps = {
+	category: string;
+	searchTerm: string;
+};
+
+export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageViewTrackerProps ) {
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+
+	// `PageViewTracker` pushes a new event only when `path` changes, so we track
+	// `calypso_pattern_library_search` to have accurate data on which search terms are used
+	useEffect( () => {
+		if ( searchTerm ) {
+			recordTracksEvent( 'calypso_pattern_library_search', {
+				is_logged_in: isLoggedIn ? '1' : '0',
+				search_term: searchTerm,
+				user_is_dev_account: isDevAccount ? '1' : '0',
+			} );
+		}
+	}, [ isDevAccount, isLoggedIn, searchTerm ] );
```

Comment:
```suggestion
	useEffect( () => {
		if ( searchTerm ) {
			recordTracksEvent( 'calypso_pattern_library_search', {
				is_logged_in: isLoggedIn ? 'true' : 'false',
				search_term: searchTerm,
				user_is_dev_account: isDevAccount ? '1' : '0',
			} );
		}

		if ( category ) {
			recordTracksEvent( 'calypso_pattern_library_filter', {
				category: category,
				is_logged_in: isLoggedIn ? 'true' : 'false',
				user_is_dev_account: isDevAccount ? '1' : '0',
			} );
		}
	}, [ isDevAccount, isLoggedIn, searchTerm, category ] );
```

Two things here:
* Let's report `is_logged_in` as `true` and `false` for consistency, since that is the data that's been collected historically. Unfortunately, that means we're treating boolean values differently for `is_logged_in` and `user_is_dev_account`, but I'd rather we maintain consistency with existing data.
* Currently, the page view event isn't tracked when user selects a category. This should address that! Let's also track the `calypso_pattern_library_filter` event. It's easier to monitor the data using our tracks MC tool with a separate events, since we have a ton of `calypso_page_view` events.

Reply:
> Let's report is_logged_in as true and false for consistency
👍

> Currently, the page view event isn't tracked when user selects a category

Oh yeah, I had a solution for this previously, but you're right that it wouldn't push a new event right now. Updating the `path` we pass to `PageViewTracker` should have the same effect.